src/app/zap-templates/templates/app/CHIPClusters-src.zapt

```diff
@@ -75,8 +76,30 @@ CHIP_ERROR {{asCamelCased name false}}Cluster::DiscoverAttributes(Callback::Canc
 CHIP_ERROR {{asCamelCased parent.name false}}Cluster::ReadAttribute{{asCamelCased name false}}(Callback::Cancelable * onSuccessCallback, Callback::Cancelable * onFailureCallback)
 {
     uint8_t seqNum = mDevice->GetNextSequenceNumber();
-    System::PacketBufferHandle encodedCommand = encode{{asCamelCased parent.name false}}ClusterRead{{asCamelCased name false}}Attribute(seqNum, mEndpoint);
-    return SendCommand(seqNum, std::move(encodedCommand), onSuccessCallback, onFailureCallback);
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    chip::app::AttributePathParams attributePath;
+    attributePath.mNodeId     = mDevice->GetDeviceId();
+    attributePath.mEndpointId = mEndpoint;
+    attributePath.mClusterId  = mClusterId;
+    attributePath.mFieldId    = {{asHex code 4}};
+    attributePath.mFlags.Set(chip::app::AttributePathParams::Flags::kFieldIdValid);
+    if (onSuccessCallback != nullptr || onFailureCallback != nullptr)
```

Comment:
I don't follow this part.  The code in `CHIPClientCallbacks.zapt ` was assuming these callbacks are both non-null assuming there are any callbacks at all, as far as I can tell.  So here we either need to check both are non-null or assume they're both non-null.  Or we need to fix that code to deal with only one callback being null.

I guess the old code had the same problem, now that I've read that far?  Followup, please.